t/ui/14-dashboard.t

```diff
@@ -94,13 +94,25 @@ is(
 
 $driver->find_element_by_link_text('400')->click();
 is($driver->find_element('#more_builds b')->get_text(), 400, 'limited to the selected number');
+like($driver->get_current_url(), qr/1001\?limit_builds=400/, 'url shows the selected build_limits');
+$driver->find_element_by_link_text('50')->click();
+like(
+    $driver->get_current_url(),
+    qr/1001\?(?!.*limit_builds&)limit_builds=50/,
```

Comment:
I *guess* this is the correct syntax. When I would have to write this check myself I'd probably just used `unlike` and `like` (so two separate checks) for what's supposed to be absent and what's supposed to be present. That would be much clearer as it wouldn't involve so much regex black-magic.

It would probably also make sense to check whether a previously selected option (e.g. `only_tagged=1`) is still present after using the pagination because your change *possibly* breaks that.

Reply:
Starting from the second part: is this covered https://github.com/os-autoinst/openQA/pull/6315/files#diff-9ad9a55405b1ec79d25f3b32f354896690e22a296b04a1e42b240a1b48a72763R109?

`1001\?(?!.*limit_builds&)limit_builds=50` says 

`1001\?limit_builds=50` is present and it is query is not extended. the combination of unlike+like sounds more complicated to me. in additional, more check, less faster (even is so trivial in general). it would look like, right?

```
unlike $driver->get_current_url(), qr/1001\?limit_builds=400&limit_builds=50/;
like $driver->get_current_url(), qr/1001\?limit_builds=50/;
```

Reply:
Right, the check for another option like `only_tagged=1` is indeed covered by the diff/line you've mentioned.

---

Yes, it would look like that except that I'd remove the `limit_builds=50` part from the `unlike` check.

If you don't invoke `$driver->get_current_url()` twice (but save that in a variable) the shouldn't be much of a performance overhead due to checking this twice. And in any case the performance difference is negligible.

Reply:
Why `limit_builds&`? `limit_builds` without a `=value` would not appear in the query anyway, would it?

Reply:
```perl
my @urls = (
    "/1001?limit_builds=0&limit_builds=400&limit_builds=50",
    "/1001?limit_builds=50",
    "/1001?xxx=yyy&limit_builds=50",
);
for (@urls) {
    say "==== $_ ====";
    say m/1001\?(?!.*limit_builds&)limit_builds=50/ ? "match" : "no match"
}
__END__
==== /1001?limit_builds=0&limit_builds=400&limit_builds=50 ====
no match
==== /1001?limit_builds=50 ====
match
==== /1001?xxx=yyy&limit_builds=50 ====
no match
```
So basically your regex is ensuring the same as `m/1001\?limit_builds=50/`

Reply:
nice catch @perlpunk . Lets do it as @Martchus suggested anyway